homeassistant/components/notion/binary_sensor.py

```diff
@@ -75,19 +75,21 @@ class NotionBinarySensor(NotionEntity, BinarySensorEntity):
     """Define a Notion sensor."""
 
     @callback
+    @async_guard_against_missing_task
     def _async_update_from_latest_data(self) -> None:
         """Fetch new state data for the sensor."""
-        task = self.coordinator.data["tasks"][self._task_id]
+        task = self.coordinator.data["tasks"][self.task_id]
```

Comment:
If the task ID is missing, we should set `available` to False.

Reply:
We do (in `NotionEntity`):

https://github.com/home-assistant/core/blob/f693c8a9fd9fc8449ec6588467006bd165048989/homeassistant/components/notion/__init__.py#L168-L174

However, the availability calculation doesn't stop coordinator listeners from being called (which is where this issue comes from):

https://github.com/home-assistant/core/blob/f693c8a9fd9fc8449ec6588467006bd165048989/homeassistant/helpers/update_coordinator.py#L164-L193

Reply:
Right. We call because we want every impacted entity to be marked as unavailable.

If we just put a guard here, `if self.task_id not in self.coordinator.data: return`, then the available property will return `False` and we will never end up calling `is_on` etc.